lib/SPIRV/libSPIRV/SPIRVInstruction.h

```diff
@@ -538,9 +538,13 @@ class SPIRVStore : public SPIRVInstruction, public SPIRVMemoryAccess {
     SPIRVInstruction::validate();
     if (getSrc()->isForward() || getDst()->isForward())
       return;
-    assert(getValueType(PtrId)->getPointerElementType() ==
-               getValueType(ValId) &&
-           "Inconsistent operand types");
+    if (getValueType(PtrId)->getPointerElementType() != getValueType(ValId)) {
```

Comment:
No need to wrap assert with an `if`: I mean this might be needed if assertion is not correct under some circumstances, but in that case, please wrap everything into `#ifndef NDEBUG`